src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -98,6 +98,9 @@ private static IVecInt toVecInt(IntStream stream) {
 		for (ModCandidateSet mcs : modCandidateSetMap.values()) {
 			Collection<ModCandidate> s = mcs.toSortedSet();
 			modCandidateMap.put(mcs.getModId(), s);
+			for (String modAlias : mcs.getModAliases()) {
+				modCandidateMap.put(modAlias, s);
+			}
```

Comment:
I'm not sure if this is sufficient to resolve a normal set of mods - what happens if there is an older version of fabric-api (with an id of "fabric") in the mods folder as well as a newer one (with an id of "fabric-api" and an alias of "fabric")? Depending on the order of the hash map I think this might either load both or only one? (Also this doesn't seem to have duplication checking in `FabricLoader.addMod()` - I don't think that's enough to fix this, but it would be able to at least detect when this happens)

Reply:
(Just FYI https://github.com/AMereBagatelle/fabric-loader/pull/1 seems to fix this from my testing) - this doesn't happen any more for simple mode after https://github.com/FabricMC/fabric-loader/pull/334/commits/635f5c03ac8c1cb8355d7dd33b5dc2c3989bf8b7)

Reply:
Merged the PR into the PR, thank you